### PR TITLE
[6251] Check trainee record page placement details section

### PR DESCRIPTION
**Pull request by defong**
### Context
Check trainee record page placement details section

### Changes proposed in this pull request
Show the differentiating placement detail views

### Guidance to review

Create an applicable draft trainee

And you should be able to manipulate by adding or removing the placements or simply transverse the service to give one of the following views.

http://localhost:5000/view_components/sections/view/continue_sections_placements

image

http://localhost:5000/view_components/sections/view/continue_sections_placements_validated

image

http://localhost:5000/view_components/sections/view/start_sections_placements
image

http://localhost:5000/view_components/sections/view/start_sections_placements_validated
image

http://localhost:5000/view_components/placement_details/view

default
image
image

one placements
image
image

two placements
image

image

five placements

image

image

### Important business
NOPE

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/forms/submissions/base_validator.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@ def validator(name, options)
     validator :schools, form: "Schools::FormValidator", if: :requires_schools?
     validator :funding, form: "Funding::FormValidator", if: :requires_funding?
     validator :iqts_country, form: "IqtsCountryForm", if: :requires_iqts_country?
+    validator :placements, form: "PlacementDetailForm", if: :requires_placements?
```
Comment (darokel) [inline app/forms/submissions/base_validator.rb:25]: We'll need to use a composite validator here to check `PlacementDetailForm` and the one Steve is building. Just making a note here so we don't forget!

**Review by darokel — APPROVED** [review]
Nice 👍

**Comment by defong** [comment]
[Check trainee record page - placement details section](https://trello.com/c/f95cFeYL/6251-check-trainee-record-page-placement-details-section)

**Review by stevehook — APPROVED** [review]

File: app/forms/submissions/base_validator.rb (hunk 1 above)
Comment (defong) [reply app/forms/submissions/base_validator.rb:25]: Added to snag list

**Pull request merged**